src/core/plugins/memory-plugin.ts

```diff
@@ -99,8 +128,74 @@ class MemoryPlugin extends CallablePlugin {
     }
   }
 
-  async readLocalMem(key) {
-    return await this.redisGet(`${this.keyPrefix}-${key}`);
+  async writeGlobalMem(key: string, value: string, ttl: number=0, options:MemWriteOptions={}) {
+    if(typeof value !== "string")
+      throw `Only string data can be saved.`
+
+    let memWrite: MemWrite = {
+      type: "global",
+      owner: process.env.SIGN_WALLET_ADDRESS!,
+      timestamp: getTimestamp(),
+      ttl,
+      key,
+      value,
+    }
+    return PromiseLib.resolveN(
+      this.netConfigs.tss.threshold,
+      this.nodeManager.filterNodes({isDeployer: true}).map(({peerId, id}) => {
+        return (
+          id === this.currentNodeInfo?.id
+          ?
+          this.__writeGlobal(memWrite, this.currentNodeInfo)
+          :
+          this.remoteCall(
+            peerId,
+            RemoteMethods.WriteGlobal,
+            memWrite,
+            {timeout: 5000},
+          )
+        )
+      })
+    )
+  }
+
+  async readGlobalMem(key: string): Promise<string|null> {
+    let results:string[] = await PromiseLib.resolveN(
+      this.netConfigs.tss.threshold,
+      this.nodeManager.filterNodes({isDeployer: true}).map(({peerId, id}) => {
+        return (
+          id === this.currentNodeInfo!.id
+          ?
+          this.__readGlobal(key, this.currentNodeInfo!)
+          :
+          this.remoteCall(
+            peerId,
+            RemoteMethods.ReadGlobal,
+            key,
+            {timeout: 5000},
+          )
+        )
+      }),
+      true
+    )
+    let uniqueResults = _.uniq(results.filter(r => !!r));
+    if(uniqueResults.length !== 1)
+      return null;
+    return uniqueResults[0];
+  }
+
+  @remoteMethod(RemoteMethods.WriteGlobal)
+  async __writeGlobal(memWrite: MemWrite, callerInfo: MuonNodeInfo) {
```

Comment:
Every node can call this function. It should be restricted.
1- throw err on non deployers
2- validate app(caller should be in app context)
3- validate request(how?)